internal/testbed/integration/combinedload/e2e_test.go

```diff
@@ -0,0 +1,100 @@
+//go:build e2e
+
+package combinedload
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/Dynatrace/dynatrace-otel-collector/internal/testcommon/k8stest"
+	oteltest "github.com/Dynatrace/dynatrace-otel-collector/internal/testcommon/oteltest"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+)
+
+func TestLoad_Combined(t *testing.T) {
+	testDir := filepath.Join("testdata")
+
+	k8sClient, err := k8stest.NewK8sClient()
+	require.NoError(t, err)
+
+	// Create the namespace specific for the test
+	nsFile := filepath.Join(testDir, "namespace.yaml")
+	buf, err := os.ReadFile(nsFile)
+	require.NoErrorf(t, err, "failed to read namespace object file %s", nsFile)
+	nsObj, err := k8stest.CreateObject(k8sClient, buf)
+	require.NoErrorf(t, err, "failed to create k8s namespace from file %s", nsFile)
+
+	testNs := nsObj.GetName()
+	defer func() {
+		require.NoErrorf(t, k8stest.DeleteObject(k8sClient, nsObj), "failed to delete namespace %s", testNs)
+	}()
+
+	// start up metrics-server
+	t.Log("deploying metrics-server...")
+	err = k8stest.PerformOperationOnYAMLFiles(k8sClient, filepath.Join(testDir, "metrics-server"), k8stest.CreateObject)
+	require.NoErrorf(t, err, "failed to create k8s metrics server")
+
+	defer func() {
+		err = k8stest.PerformOperationOnYAMLFiles(k8sClient, filepath.Join(testDir, "metrics-server"), k8stest.DeleteObjectFromManifest)
+		require.NoErrorf(t, err, "failed to delete k8s metrics server")
+	}()
+
+	// wait for metrics server to be ready
+	err = k8stest.WaitForDeploymentPods(k8sClient.DynamicClient, "kube-system", "metrics-server", 2*time.Minute)
+	require.NoErrorf(t, err, "failed to rollout k8s metrics server")
+	t.Log("metrics-server deployed")
+
+	// get metrics client
+	metricsClientSet, err := k8stest.NewMetricsClientSet()
+	require.NoError(t, err)
+
+	tracesConsumer := new(consumertest.TracesSink)
+	shutdownSinks := oteltest.StartUpSinks(t, oteltest.ReceiverSinks{Traces: tracesConsumer})
+	defer shutdownSinks()
+
+	testID := uuid.NewString()[:8]
+	collectorObjs := k8stest.CreateCollectorObjects(t, k8sClient, testID, filepath.Join(testDir, "collector"))
+
+	createTeleOpts := &k8stest.TelemetrygenCreateOpts{
+		ManifestsDir: filepath.Join(testDir, "telemetrygen"),
+		TestID:       testID,
+		OtlpEndpoint: fmt.Sprintf("otelcol-%s.%s:4317", testID, testNs),
+		DataTypes:    []string{"traces"},
+	}
+	t.Log("deploying telemetrygen...")
+	telemetryGenObjs, telemetryGenObjInfos := k8stest.CreateTelemetryGenObjects(t, k8sClient, createTeleOpts)
+	defer func() {
+		for _, obj := range append(collectorObjs, telemetryGenObjs...) {
+			require.NoErrorf(t, k8stest.DeleteObject(k8sClient, obj), "failed to delete object %s", obj.GetName())
+		}
+	}()
+
+	for _, info := range telemetryGenObjInfos {
```

Comment:
Where do the `telemetryGenObjInfos` come from? I assume there is one for each datatype, is that right?

Reply:
we are using a library for deploying tememetry generators - using a k8s client to apply manifests and retrieve info about the applied deployments, so we can delete it after the test is done. In this step we are waiting for all the telemetrygen object to be rolled out (we get the info about the objects on line 70)

And yes there is one for each datatype, that's why we need to use a for loop to wait for all of them to be rolled out

Reply:
But we don't get any information about which one was started? I just thought it would be cool to see in the logs "telemetrygen - metrics - started", "telemetrygen - logs - started", etc. but if it's complicated it's not worth the effort

Reply:
The starting process is handled in the library creating the telemetrygen objects [here](https://github.com/Dynatrace/dynatrace-otel-collector/blob/main/internal/testcommon/k8stest/telemetrygen.go#L35). If we want to have a log line that the telemeterygen started, I will add it directly to the library

Reply:
added some additional logging [here](https://github.com/Dynatrace/dynatrace-otel-collector/pull/304/commits/8d8d10c93c28d8dff27b2034e8b08eeb79a4f205)

You can see the logs [here](https://github.com/Dynatrace/dynatrace-otel-collector/actions/runs/11028834503/job/30629956435?pr=304)

Reply:
see here the resulting loging https://github.com/Dynatrace/dynatrace-otel-collector/pull/304/files#diff-1636c076f509ae7cc52a0b5bdae0b35130c876b80ae30c37a21770214ad59f86R23